openai_generate.py

```diff
@@ -0,0 +1,39 @@
+from joblib import Memory
+from tenacity import (
+    retry,
+    stop_after_attempt,
+    wait_random_exponential,
+)
+from openai import OpenAI
+from model_config import OPENAI_MODEL
+from typing import List
+
+memory = Memory("cache/openai", verbose=0)
+
+
+# @memory.cache
```

Comment:
For the OpenAI cache, I realized it's actually a bad idea since we want a different response for even the same prompt. Caching function has been implemented but commented out.